src/hardware/signal_generator.rs

```diff
@@ -158,20 +167,28 @@ impl SignalGenerator {
     pub fn update_waveform(&mut self, new_config: Config) {
         self.config = new_config;
     }
+
+    /// Clear the phase accumulator.
+    pub fn clear_phase_accumulator(&mut self) {
+        self.phase_accumulator = 0;
+    }
 }
 
 impl core::iter::Iterator for SignalGenerator {
     type Item = i16;
 
     /// Get the next value in the generator sequence.
     fn next(&mut self) -> Option<i16> {
-        let sign = self.phase_accumulator.is_negative();
+        let phase = self
+            .phase_accumulator
+            .wrapping_add(self.config.phase_offset);
+        let sign = phase.is_negative();
         self.phase_accumulator = self
             .phase_accumulator
-            .wrapping_add(self.config.frequency_tuning_word[sign as usize]);
+            .wrapping_add(self.config.phase_increment[sign as usize]);
 
         let scale = match self.config.signal {
-            Signal::Cosine => (idsp::cossin(self.phase_accumulator).0 >> 16),
+            Signal::Cosine => (idsp::cossin(phase).0 >> 16),
```

Comment:
Don't we also have to modify `Signal::Triangle` to use `phase` instead of `self.phase_accumulator`?